File: app/src/main/java/me/pwcong/jpstart/comparator/JPItemComparator.java
package me.pwcong.jpstart.comparator;

import java.util.Comparator;

import me.pwcong.jpstart.mvp.bean.JPItem;

/**
 * Created by Pwcong on 2016/9/28.
 */

public class JPItemComparator implements Comparator<JPItem> {

    @Override
    public int compare(JPItem o1, JPItem o2) {

        if (o1.getRow() < o2.getRow()) {
            return -1;
        } else if (o1.getRow() == o2.getRow()) {

            if (o1.getColumn() < o2.getColumn()) {
                return -1;
            } else if (o1.getColumn() == o2.getColumn()) {
                return 0;
            } else {
                return 1;
            }

        } else {
            return 1;
        }
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/comparator/JPItemComparator.kt
package com.github.pwcong.jpstart.comparator

import com.github.pwcong.jpstart.mvp.bean.JPItem

class JPItemComparator : Comparator<JPItem?> {
    override fun compare(o1: JPItem?, o2: JPItem?): Int {
        return if (o1 == null || o2 == null || o1.row < o2.row) {
            -1
        } else if (o1.row == o2.row) {
            if (o1.column < o2.column) {
                -1
            } else if (o1.column == o2.column) {
                0
            } else {
                1
            }
        } else {
            1
        }
    }
}